### ISSUE TITLE
Programmatic use: "additional options" aren't additional

### ISSUE THREAD

**Issue by jerrygreen** [AssemblyScript/assemblyscript#2247]
In the docs, [Programmatic usage](https://www.assemblyscript.org/compiler.html#programmatic-usage) section, this code is mentioned:

```js
  // Additional API options
  stdout?: ...,
  stderr?: ...,
  readFile?: ...,
  writeFile?: ...,
  listFiles?: ...,
  reportDiagnostic?: ...,
  transforms?: ...
```

When I try the simplest code like this:

```js
asc.main(['myModule.ts'])
```

I'm getting errors then:

```txt
Error: 'options.stdout' must be specified
```

When I pass `options.stdout`, it raises error about `stderr` requirement, then `readFile`, etc.

Those options, though called "additional"... Aren't additional.

Due to the issue #2241, I'm not using the most recent version and instead I'm using `assemblyscript@0.19.23`, but this "additional" problem seems to be around for an eternity anyway, and I guess it is present in the recent versions, too.

Please make this programmatic usage to be what it's shown to be: make additional options truly additional. Thanks.

**Comment by dcodeIO** [comment]
This is fixed in 0.20, where `stdout` and `stderr` now default to a memory stream.

**Comment by jerrygreen** [comment]
Ok, then I'm closing this one.
